src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -668,6 +741,156 @@ InferenceEngine::Precision FullyConnected::getRuntimePrecision() const {
+    normalizedWeightDims = {static_cast<dnnl::memory::dim>(weightDims[0]),
+                            static_cast<dnnl::memory::dim>(weightDims[1]),
+                            dnnl::memory::dim{1},
+                            dnnl::memory::dim{1}};
+    auto convWeightDescAny = dnnl::memory::desc(normalizedWeightDims, wdt, dnnl::memory::format_tag::any);
+
+    std::shared_ptr<dnnl::convolution_forward::desc> desc;
+    if (withBiases) {
+        desc = std::make_shared<dnnl::convolution_forward::desc>(prop_kind::forward_scoring, dnnl::algorithm::convolution_direct,
+                                convInDesc, convWeightDescAny, biasDesc, convOutDesc,
+                                dnnl::memory::dims{1, 1},   // stride
+                                dnnl::memory::dims{0, 0},   // dilation
+                                dnnl::memory::dims{0, 0},   // paddingL
+                                dnnl::memory::dims{0, 0});  // paddingR
+    } else {
+        desc = std::make_shared<dnnl::convolution_forward::desc>(prop_kind::forward_scoring, dnnl::algorithm::convolution_direct,
+                                convInDesc, convWeightDescAny, convOutDesc,
+                                dnnl::memory::dims{1, 1},   // stride
+                                dnnl::memory::dims{0, 0},   // dilation
+                                dnnl::memory::dims{0, 0},   // paddingL
+                                dnnl::memory::dims{0, 0});  // paddingR
+    }
+
+    return DnnlDesriptor(desc);
+}
+
+bool FullyConnected::canBeExecutedInConv1x1() const {
+    bool shouldTry = false;
+    const auto inRank = getInputShapeAtPort(DATA_ID).getMaxDims().size();
+    const auto weightRank = getInputShapeAtPort(WEIGHTS_ID).getStaticDims().size();
+    // disable rank=4: rank == 4 means a matrix: N * IC * H * W --> N * (IC*H*W) --> N * IC * 1 * 1, it should not be efficient which acts as vector multiply
+    if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core) &&
+        getOriginalInputPrecisionAtPort(DATA_ID) == InferenceEngine::Precision::FP32 &&
+        one_of(inRank, 2, 3) && weightRank == 2) {
+        auto dstMemPtr = getChildEdgesAtPort(0)[0]->getMemoryPtr();
+        DnnlMemoryDescCPtr outDesc = dstMemPtr->GetDescWithType<DnnlMemoryDesc>();
+        // brg convolution does not support stride
+        if (outDesc->getDnnlDesc().data.offset0 == 0)
+            shouldTry = true;
+    }
+
+    return shouldTry;
+}
+
+FullyConnected::ExecutorInnerProduct::ExecutorInnerProduct(const dnnl::inner_product_forward::primitive_desc& pd) {
+    execPrim.reset(new dnnl::inner_product_forward(pd));
+}
+
+FullyConnected::ExecutorConv1x1::ExecutorConv1x1(const dnnl::convolution_forward::primitive_desc& pd) {
+    execPrim.reset(new dnnl::convolution_forward(pd));
+    srcDesc = pd.src_desc();
+    weightDesc = pd.weights_desc();
+    dstDesc = pd.dst_desc();
+    implementationType = parse_impl_name(pd.impl_info_str());
+}
```

Comment:
Could you please elaborate on why `DnnlExecutor` base class is used? It seems that no functionality from this class is being used in these derivatives.

Reply:
We need a container to encapsulate the primitive, src_desc etc. `DnnlExecutor` is a good choice.